libraries/Microsoft.Bot.Builder.AI.Orchestrator/OrchestratorAdaptiveRecognizer.cs

```diff
@@ -0,0 +1,280 @@
+using Microsoft.Bot.Builder.TraceExtensions;
+using Microsoft.Bot.Schema;
+using Microsoft.Orchestrator;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Bot.Builder.AI.Orchestrator
+{
+    public class OrchestratorAdaptiveRecognizer : Recognizer
+    {
+        [JsonProperty("$kind")]
+        public const string Kind = "Microsoft.OrchestratorRecognizer";
+
+        /// <summary>
+        /// Intent name that will be produced by this recognizer if the child recognizers do not have consensus for intents.
+        /// </summary>
+        private const string ChooseIntent = "ChooseIntent";
+
+        /// <summary>
+        /// Property name for candidate intents that meet the ambiguity threshold.
+        /// </summary>
+        private const string CandidatesCollection = "candidates";
+
+        private static string modelPath = null;
+        private string snapshotPath = null;
+        private OrchestratorRecognizer recognizer = null;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="OrchestratorAdaptiveRecognizer"/> class.
+        /// </summary>
+        /// <param name="callerLine">caller line.</param>
+        /// <param name="callerPath">caller path.</param>
+        [JsonConstructor]
+        public OrchestratorAdaptiveRecognizer([CallerFilePath] string callerPath = "", [CallerLineNumber] int callerLine = 0)
+            : base(callerPath, callerLine)
+        {
+        }
+
+        /// <summary>
+        /// Gets or sets the full path to the NLR model to use.
+        /// </summary>
+        /// <value>
+        /// Model path.
+        /// </value>
+        [JsonProperty("modelPath")]
+        public StringExpression ModelPath { get; set; }
+
+        /// <summary>
+        /// Gets or sets the full path to the snapshot to use.
+        /// </summary>
+        /// <value>
+        /// Snapshot path.
+        /// </value>
+        [JsonProperty("snapshotPath")]
+        public StringExpression SnapshotPath { get; set; }
+
+        /// <summary>
+        /// Gets or sets the entity recognizers.
+        /// </summary>
+        /// <value>
+        /// The entity recognizers.
+        /// </value>
+        [JsonProperty("entityRecognizers")]
+#pragma warning disable CA2227 // Collection properties should be read only (keeping this consistent with RegexRecognizer)
+        public List<EntityRecognizer> EntityRecognizers { get; set; } = new List<EntityRecognizer>();
+#pragma warning restore CA2227 // Collection properties should be read only
+
+        /// <summary>
+        /// Gets or sets the disambiguation score threshold.
+        /// </summary>
+        /// <value>
+        /// Recognizer returns ChooseIntent (disambiguation) if other intents are classified within this score of the top scoring intent.
+        /// </value>
+        [JsonProperty("disambiguationScoreThreshold")]
+        public NumberExpression DisambiguationScoreThreshold { get; set; } = 0.05F;
```

Comment:
>DisambiguationScoreThreshold [](start = 32, length = 28)

Are we going to build this into all of our recognizers?

Reply:
Moved to Recognizer base. Thanks Tom! :)